MAVProxy/mavproxy.py

```diff
@@ -1112,7 +1115,7 @@ def quit_handler(signum = None, frame = None):
     (mpstate.status.logdir, logpath_telem, logpath_telem_raw) = log_paths()
 
     for module in opts.load_module:
-        modlist = module.split(',')
+        modlist = shlex.split(module, ',')
```

Comment:
The module name spiting logic separates the JSON into two modules one called `RTCM:{"addr":"bla"` and one called `"inst:"2"}` , hence the `shlex` call

Reply:
IMO rather than use `shlex.split()` here I think we should do a json argument detection. That would save the json formatting and keep the existing downstream code as it it

Reply:
What exactly do you mean with " json argument detection"?

Reply:
Replace module.split(',') with a function that is looking for json strings within the module string and avoids splitting them. A regular expression could be used to identify json substrings within the module string. Possibly something like `([:{].*?[}])` (note: untested)
Then split the string into a list knowing to avoid any split locations that are in a json substring. That's one way of doing it anyway...

Reply:
A quick play gives `,+(?=[^{}]*(?:\{|$))` as a working solution

so
```
regex = r",+(?=[^{}]*(?:\{|$))"
modlist = re.split(regex , module)
```
should work

see: https://regex101.com/r/uN8oK4/1